jupyter/restore_raft_quorum_with_tablets.ipynb

```diff
@@ -0,0 +1,680 @@
+   "source": [
+    "# Recovery procedure\n",
+    "Execute each step one by one\n",
+    "Use 'GET CLUSTER STATUS' cell to check cluster status between steps if need"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f632964f",
+   "metadata": {},
+   "source": [
+    "## Step 2.\n",
+    "Stop one of alive node in alive dc.\n",
+    "Assume that it is the latest node in list"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "1f1893f8",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "stopped_alive_node = nodes_by_region[alive_dc_region][-1]\n",
+    "stopped_node_host_id = stopped_alive_node.host_id\n",
+    "stopped_alive_node.stop_scylla()\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a7c8a1ce",
+   "metadata": {},
+   "source": [
+    "## Step 4.\n",
+    "Choose alive nodes which will be used for cluster restore\n",
+    "Rolling restart of alive nodes"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "42acf42c",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "base_alive_nodes = nodes_by_region[alive_dc_region][:-1]\n",
+    "cluster.restart_scylla(nodes=base_alive_nodes)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "b4fc3063",
+   "metadata": {},
+   "source": [
+    "## Step 6. \n",
+    "Find recovery coordinator"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "7a5c9943",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "result = verification_node.run_cqlsh(\"select value from system.scylla_local where key = 'raft_group0_id'\", split=True)\n",
+    "group0_id = result[3]\n",
+    "commit_idx = 0\n",
+    "recovery_coordinator = None\n",
+    "for node in base_alive_nodes:\n",
+    "    cidx = verification_node.run_cqlsh(f\"select commit_idx from system.raft where group_id = {group0_id}\", split=True)[3]\n",
+    "    if int(cidx) > commit_idx:\n",
+    "        commit_idx = int(cidx)\n",
+    "        recovery_coordinator = node\n",
+    "\n",
+    "print(recovery_coordinator.name, recovery_coordinator.ip_address, recovery_coordinator.host_id)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f076d21c",
+   "metadata": {},
+   "source": [
+    "## Step 7. \n",
+    "Remove raft_group_id from system.scylla_local and truncate system.discovery on each live node."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "c8156024",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "for node in base_alive_nodes:\n",
+    "    node.run_cqlsh(\"DELETE value FROM system.scylla_local WHERE key = 'raft_group0_id'\")\n",
+    "    node.run_cqlsh(\"TRUNCATE system.discovery\")\n",
+    "    "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2ed22247",
+   "metadata": {},
+   "source": [
+    "## Step 8.\n",
+    "Set the new scylla.yaml parameter, recovery_leader, to Host ID of the recovery leader on each live node"
```

Comment:
We must also send the `SIGHUP` signal to all ScyllaDB processes to ensure that the changed parameter is applied (see https://docs.scylladb.com/manual/master/troubleshooting/handling-node-failures.html#manual-recovery-procedure, step 5).